aten/src/ATen/native/transformers/sdp_utils_cpp.h

```diff
@@ -407,9 +408,10 @@ inline bool check_batch_size_and_num_heads_dense(sdp_params const& params, bool
   }
 
   if(params.enable_gqa && supports_gqa){
-    return check_grouped_query_attention(params, debug);
+    return check_grouped_query_attention<requires_same_num_heads>(params, debug);
   }
 
+  // same num heads condition for non-gqa case
```

Comment:
Has this not also been fixed in recent CUDNN versions?

Reply:
Not sure I understand this question---this comment is basically saying that if gqa isn't enabled (via the argument to the SDPA call) or if the backend doesn't support it (which cuDNN does) it goes to the original check that the number of head is the same for each case (non-gqa case). For cuDNN as with the other backends we will error-out here as the user didn't explicitly enable GQA.

This PR as a whole isn't enabling any new functionality, it's restoring some previous functionality that I inadvertently removed in #149976 when switching to (this) GQA check that the other backends used without realizing that it was stricter than what cuDNN supported.